Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/InputArgument.java

```diff
@@ -0,0 +1,83 @@
+package nl.utwente.group10.ui.components.blocks.function;
+
+import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.ui.components.ComponentLoader;
+import nl.utwente.group10.ui.components.anchors.InputAnchor;
+import nl.utwente.group10.ui.components.blocks.Block;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Pane;
+
+/**
+ * A class that represents an input field inside a FunctionBlock.
+ * This basically combines a label with an anchor to which an input can be connected.
+ */
+public class InputArgument extends Pane implements ComponentLoader{
+    
+    /** The label on which to display type information. */
+    @FXML Label inputLabel;
+    
+    /** The property of the text to be displayed on the inputLabel. */
+    private StringProperty inputText;
+    
+    /** The InputAnchor belonging to this InputArgument */
+    private InputAnchor inputAnchor;
+    
+    /**
+     * Constructs a new InputArgument.
+     * @param block Block to which this InputArgument belongs.
+     * @param signature Type signature as should be accepted by the InputAnchor.
+     */
+    public InputArgument(Block block, Type signature) {
+        inputText = new SimpleStringProperty(signature.toHaskellType());
+        this.loadFXML("InputArgument");
+        
+        inputAnchor = new InputAnchor(block, signature);
+        inputAnchor.layoutXProperty().bind(inputLabel.widthProperty().divide(2));
+        inputAnchor.isErrorProperty().addListener((ObservableValue<? extends Boolean> value, Boolean oldValue, Boolean newValue) -> setError(newValue));
```

Comment:
The lambda signature is a bit long... Would it make sense to use a method reference here instead?

````
inputAnchor.isErrorProperty().addListener(this::setError); 

// ...

private void setError(ObservableValue<? extends Boolean> value, Boolean oldValue, Boolean newValue) {
    setError(newValue)
}

// ...
````